lib/puppet/indirector/file_bucket_file/file.rb

```diff
@@ -218,7 +218,7 @@ def path_for(bucket_path, digest, subfile = nil)
     #   and content as that in the bucket_file
     # @api private
     def verify_identical_file(contents_file, bucket_file)
-      (bucket_file.size == Puppet::FileSystem.size(contents_file)) &&
+      (bucket_file.to_binary.bytesize == Puppet::FileSystem.size(contents_file)) &&
```

Comment:
Note that while #to_binary here sounds like it's returning something with binary encoding, all it's actually doing (if this is the in-memory StringContents representation) is returning the contents as-is.